### ISSUE TITLE
% formatting fails to find formatting code in bytes type after a null byte

### ISSUE THREAD

**Issue by cb60ce86-4103-42f8-8f08-67a4bb264215** [python/cpython#74120]
BPO | [29934](https://bugs.python.org/issue29934)
--- | :---
Nosy | @zhangyangyu
Superseder | <li>bpo-29714: can&#x27;t interpolate byte string with \x00 before replacement identifier</li>

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = <Date 2017-03-28.14:43:07.948>
created_at = <Date 2017-03-28.14:37:41.313>
labels = ['interpreter-core', 'type-bug']
title = '% formatting fails to find formatting code in bytes type after a null byte'
updated_at = <Date 2017-03-28.14:47:13.514>
user = 'https://bugs.python.org/boramalper'
```

bugs.python.org fields:
```python
activity = <Date 2017-03-28.14:47:13.514>
actor = 'xiang.zhang'
assignee = 'none'
closed = True
closed_date = <Date 2017-03-28.14:43:07.948>
closer = 'xiang.zhang'
components = ['Interpreter Core']
creation = <Date 2017-03-28.14:37:41.313>
creator = 'boramalper'
dependencies = []
files = []
hgrepos = []
issue_num = 29934
keywords = []
message_count = 3.0
messages = ['290724', '290726', '290727']
nosy_count = 2.0
nosy_names = ['boramalper', 'xiang.zhang']
pr_nums = []
priority = 'normal'
resolution = 'duplicate'
stage = 'resolved'
status = 'closed'
superseder = '29714'
type = 'behavior'
url = 'https://bugs.python.org/issue29934'
versions = ['Python 3.6']
```

</p></details>

**Comment by cb60ce86-4103-42f8-8f08-67a4bb264215** [comment]
Hello,

In Python 3.6.0, % formatting fails to find formatting code after a null byte in bytes type.

```pycon
Example:
    >>> "%s_\x00%s" % ("hello", "world")
    'hello_\x00world'
    >>> b"%s_\x00%s" % (b"hello", b"world")
    Traceback (most recent call last):
      File "<stdin>", line 1, in <module>
    TypeError: not all arguments converted during bytes formatting
```


In contrast, the exact same code works as expected in Python 3.5:

```pycon
    >>> "%s_\x00%s" % ("hello", "world")
    'hello_\x00world'
    >>> b"%s_\x00%s" % (b"hello", b"world")
    b'hello_\x00world'
```


I used Python 3.6.0 that I installed using pyenv 1.0.8 on Kubuntu 16.04 x86_64.

**Comment by zhangyangyu** [comment]
Yes, this is a regression in 3.6.0 and it has been fixed in bpo-29714 for 3.6.1. Try the new version. :-)

**Comment by zhangyangyu** [comment]
Sorry, the fix is for 3.6.2 so 3.6.1 would still suffer from it. :-(
